libs/designer-ui/src/lib/createConnection/index.tsx

```diff
@@ -583,29 +615,47 @@ export const CreateConnection = (props: CreateConnectionProps): JSX.Element => {
-              const connectionParameterProps: ConnectionParameterProps = {
-                parameterKey: key,
-                parameter,
-                value: parameterValues[key],
-                setValue: (val: any) => setParameterValues({ ...parameterValues, [key]: val }),
-                isSubscriptionDropdownDisabled: gatewayServiceConfig?.disableSubscriptionLookup,
-                isLoading,
-                selectedSubscriptionId,
-                selectSubscriptionCallback,
-                availableGateways,
-                availableSubscriptions,
-              };
-
-              const customParameterOptions = ConnectionParameterEditorService()?.getConnectionParameterEditor({
-                connectorId: connectorId,
-                parameterKey: key,
-              });
-              if (customParameterOptions) {
-                const CustomConnectionParameter = customParameterOptions.EditorComponent;
-                return <CustomConnectionParameter key={key} {...connectionParameterProps} />;
+              const mappingName = parameter?.uiDefinition?.credentialMapping?.mappingName;
+              if (mappingName) {
+                // This parameter belongs to a mapping.
+                if (!allParameterMappings.has(mappingName)) {
+                  allParameterMappings.add(mappingName);
+                  // This is the first time this mapping has been encountered,
+                  // we need to check if there is an Editor for it - overriding each included parameters.
+                  const parameters = fromPairs(
+                    Object.entries(capabilityEnabledParameters).filter(
+                      ([_, parameter]) => parameter.uiDefinition?.credentialMapping?.mappingName === mappingName
+                    )
+                  );
+
+                  const credentialMappingOptions = ConnectionParameterEditorService()?.getCredentialMappingEditorOptions?.({
+                    connectorId,
+                    mappingName,
+                    parameters,
+                  });
+
+                  if (credentialMappingOptions) {
+                    activeParameterMappings.add(mappingName);
+                    const CredentialsMappingEditorComponent = credentialMappingOptions.EditorComponent;
+                    const props: IConnectionCredentialMappingEditorProps = {
+                      connectorId,
+                      mappingName,
+                      parameters,
+                      setParameterValues,
+                      renderParameter,
+                    };
+                    return <CredentialsMappingEditorComponent key={mappingName} {...props} />;
+                  }
+                }
+
+                // If we encounter an already active mapping,
+                // we skip the parameter rendering since an Editor was already found and rendered.
+                if (activeParameterMappings.has(mappingName)) {
+                  return null;
+                }
```

Comment:
Can we switch the logic for the default parameters vs mapping parameters here around?  It looks like you moved our default logic apps code out to a different spot, I'd rather you do that for the mapping code.
We can leave our existing parameter code in place and just have it run `if (mappingName) return renderMappingParameter(etc)` before and return the special case.  No need to move what we have out to a new spot if it's still the "default" behavior.